### PR TITLE
Add document for custom object example

### PR DESCRIPTION
**Pull request by Taost**
The document introduces the custom object example. It also provides steps to configure a custom object.

### CODE REVIEW

**Review by AdamLiechty — APPROVED** [review]

File: docs/CustomObject.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Custom Object
+
+[The custom object example](../force-app/main/default/objects/Visit__c) demonstrates the capability to work with custom objects in mobile offline app. Like standard objects, user can view, create, and edit records of custom objects in offline.
```
Comment (khawkins) [inline docs/CustomObject.md:3]:
```suggestion
[The custom object example](../force-app/main/default/objects/Visit__c) demonstrates the capability to work with custom objects in the Mobile Offline app. Like standard objects, users can view, create, and edit records of custom objects while their device is offline.
```

File: docs/CustomObject.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+The following steps are recommended for setting up a custom object.
```
Comment (khawkins) [inline docs/CustomObject.md:5]: Am I right in assuming that these bullet points are a 1:1 mapping to the actions spelled out explicitly by example (albeit specifically for the `Visit__c` object) in the configuration section below?

File: docs/CustomObject.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+- Create a custom object with designed fields.
```
[the remark this answers is not in the document]
Comment (Taost) [reply docs/CustomObject.md:6]: Oh, the word may look ambiguous here. It is supposed to say that a custom object should be created with fields based on customer's business needs.

File: docs/CustomObject.md (hunk 2 above)
Comment (Taost) [reply docs/CustomObject.md:5]: Yes, that's correct.

File: docs/CustomObject.md (hunk 3 above)
Comment (khawkins) [inline docs/CustomObject.md:6]:
```suggestion
- Create a custom object and the associated fields needed for your use case.
```

File: docs/CustomObject.md (hunk 2 above)
Comment (khawkins) [inline docs/CustomObject.md:5]:
```suggestion
The following steps generally describe the setup of a custom object for use in offline scenarios. You can see a specific example of these steps in the "Configure" section below.
```

File: docs/CustomObject.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+- Deploy lightning web components and quick actions on the object.
```
Comment (khawkins) [inline docs/CustomObject.md:7]: Just to make sure I understand: this is essentially describing the process of creating the quick actions and associated LWC configurations (and code) for the custom object, correct?

Comment (Taost) [reply docs/CustomObject.md:7]: yeah, exactly.

File: docs/CustomObject.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
+# Custom Object
+
+[The custom object example](../force-app/main/default/objects/Visit__c) demonstrates the capability to work with custom objects in the Mobile Offline app. Like standard objects, users can view, create, and edit records of custom objects while their device is offline.
+
+The following steps generally describe the setup of a custom object for use in offline scenarios. You can see a specific example of these steps in the "Configure" section below.
+- Create a custom object and the associated fields needed for your use case.
+- Deploy lightning web components and quick actions on the object.
```
Comment (khawkins) [inline docs/CustomObject.md:7]:
```suggestion
- Configure and deploy Lightning web components and quick actions for the object.
```

File: docs/CustomObject.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+- Add quick actions to the object layout.
+- Add a custom tab on the object so that users can create object records on the browser.
+- Add or update a permission set to give users read/edit permissions to the object and its fields.
+
+> **Note**
+> Permissions for newly created custom objects are disabled by default. Admin needs to enable the object and field permissions in permission sets for users to access the custom objects. 
+
+In this example, a `Visit` custom object is configured by deploying the relevant code artifacts to your org.
```
Comment (khawkins) [inline docs/CustomObject.md:15]: I would put this under the following header, before the list of steps.

File: docs/CustomObject.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+The following steps generally describe the setup of a custom object for use in offline scenarios. You can see a specific example of these steps in the "Configure" section below.
+- Create a custom object and the associated fields needed for your use case.
+- Deploy lightning web components and quick actions on the object.
+- Add quick actions to the object layout.
+- Add a custom tab on the object so that users can create object records on the browser.
+- Add or update a permission set to give users read/edit permissions to the object and its fields.
+
+> **Note**
+> Permissions for newly created custom objects are disabled by default. Admin needs to enable the object and field permissions in permission sets for users to access the custom objects.
```
Comment (khawkins) [inline docs/CustomObject.md:13]:
```suggestion
> Permissions for newly created custom objects are disabled by default. Your org admin needs to enable the object and field permissions in permission sets for users to access the custom objects. 
```

File: docs/CustomObject.md (hunk 6 above)
Comment (Taost) [reply docs/CustomObject.md]: It was moved to the next section.

File: docs/CustomObject.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+- Add or update a permission set to give users read/edit permissions to the object and its fields.
+
+> **Note**
+> Permissions for newly created custom objects are disabled by default. Admin needs to enable the object and field permissions in permission sets for users to access the custom objects. 
+
+In this example, a `Visit` custom object is configured by deploying the relevant code artifacts to your org.
+
+## Configure Custom Object
+
+1. Deploy the following code artifacts for the `Visit` object in order.
```
Comment (khawkins) [inline docs/CustomObject.md:19]:
```suggestion
1. Deploy the following code artifacts for the `Visit` object, in the order outlined below.
```

File: docs/CustomObject.md [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    sfdx force:source:deploy -p "force-app/main/default/layouts/Visit__c-Visit Layout.layout-meta.xml"
+
+    # Add the Visits tab for user to create records on the browser
+    sfdx force:source:deploy -p force-app/main/default/tabs/Visit__c.tab-meta.xml
+
+    # Add the permission set "Offline Custom Objects" for accessing Visit object
+    sfdx force:source:deploy -p force-app/main/default/permissionsets/Offline_Custom_Objects.permissionset-meta.xml
+    ``` 
+
+2. Admin assigns the permission set "Offline Custom Objects" to org users.
```
Comment (khawkins) [inline docs/CustomObject.md:45]:
```suggestion
2. In your org, an org admin should assign the permission set `Offline Custom Objects` to org users.
```

**Review by khawkins — APPROVED** [review]
I think that's all for me! Looks good overall.

**Review by khawkins — APPROVED** [review]

**Pull request merged**